plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -929,26 +930,32 @@ func (c *PodAffinityChecker) anyPodMatchesPodAffinityTerm(pod *v1.Pod, allPods [
 	if len(term.TopologyKey) == 0 {
 		return false, false, errors.New("Empty topologyKey is not allowed except for PreferredDuringScheduling pod anti-affinity")
 	}
-	matchingPodExists := false
+
 	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(pod, term)
 	selector, err := metav1.LabelSelectorAsSelector(term.LabelSelector)
 	if err != nil {
 		return false, false, err
 	}
+
+	podNum := int32(1)
 	for _, existingPod := range allPods {
 		match := priorityutil.PodMatchesTermsNamespaceAndSelector(existingPod, namespaces, selector)
 		if match {
-			matchingPodExists = true
```

Comment:
@bsalamat , yes, it's what I'm trying to do, e.g. no modification to affinity related code. How about following suggestion?

>maybe MaxMatchingPods and MinMatchingPods is better: matching pod num is bigger than MinMatchingPods and smaller than MaxMatchingPods; if affinity term, schedule it, if anti-affinity term, do not schedule it. In this PR, we only cover MinPodMatchingPods for anti-affinity.